AMW_angular/io/src/app/settings/environments/environments-page.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, inject, OnInit } from '@angular/core';
+import { IconComponent } from '../../shared/icon/icon.component';
+import { EnvironmentService } from '../../deployment/environment.service';
+import { ToastService } from '../../shared/elements/toast/toast.service';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { Environment, EnvironmentTree } from '../../deployment/environment';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-environments-page',
+  standalone: true,
+  imports: [IconComponent],
+  templateUrl: './environments-page.component.html',
+})
+export class EnvironmentsPageComponent implements OnInit {
+  private environmentsService = inject(EnvironmentService);
+  private toastService = inject(ToastService);
+  private error$ = new BehaviorSubject<string>('');
+  private destroy$ = new Subject<void>();
+  environmentTree: EnvironmentTree[];
+
+  ngOnInit(): void {
+    this.error$.pipe(takeUntil(this.destroy$)).subscribe((msg) => {
+      msg !== '' ? this.toastService.error(msg) : null;
+    });
+    //this.getUserPermissions(); //TODO Handle Permissons
+    this.environmentsService.getContexts().subscribe((contexts) => {
+      this.environmentTree = this.buildEnvironmentTree(contexts)[0].children;
+    });
+  }
```

Comment:
It would be nice to convert `this.environmehtTree` to a `computed` signal.

first (not in ngOnInit) but as class property, define a signal for your contexts (the response from the server) and then use the computed funtion to get a signal for your environmnetTree:

```
contexts = toSignal(this.environmentsService.getContexts())

environmentTree = computed(() => 
    if(!this.contexts()) return // guard: wait until environments are loaded
    return this.buildEnvironmentTree(this.contexts())[0].children;
)